front/components/vaults/FoldersHeaderMenu.tsx

```diff
@@ -1,57 +1,95 @@
 import {
   Button,
   CloudArrowUpIcon,
+  Cog6ToothIcon,
   DocumentTextIcon,
   DropdownMenu,
   PlusIcon,
   TableIcon,
 } from "@dust-tt/sparkle";
+import type { DataSourceType, VaultType, WorkspaceType } from "@dust-tt/types";
 import type { RefObject } from "react";
+import { useState } from "react";
 
 import type { ContentActionsRef } from "@app/components/vaults/ContentActions";
+import VaultUpsertFolderModal from "@app/components/vaults/VaultUpsertFolderModal";
+import { useDataSources } from "@app/lib/swr";
 
 type FoldersHeaderMenuProps = {
+  owner: WorkspaceType;
+  vault: VaultType;
+  folder: DataSourceType;
   contentActionsRef: RefObject<ContentActionsRef>;
 };
 
 export const FoldersHeaderMenu = ({
+  owner,
+  vault,
+  folder,
   contentActionsRef,
 }: FoldersHeaderMenuProps) => {
+  const [showEditFolderModal, setShowEditFolderModal] = useState(false);
+
+  const { dataSources } = useDataSources(owner);
+
   return (
-    <DropdownMenu>
-      <DropdownMenu.Button>
-        <Button
-          size="sm"
-          label="Add data"
-          icon={PlusIcon}
-          variant="primary"
-          type="menu"
-        />
-      </DropdownMenu.Button>
+    <>
+      <VaultUpsertFolderModal
+        isOpen={showEditFolderModal}
+        setOpen={(isOpen) => {
+          setShowEditFolderModal(isOpen);
+        }}
+        owner={owner}
+        vault={vault}
+        dataSources={dataSources}
+        folder={folder}
+      />
```

Comment:
Maybe we could move it to ContentActions ?

Reply:
Discussed IRL, not to do for this one!